networkx/algorithms/approximation/steinertree.py

```diff
@@ -0,0 +1,76 @@
+import networkx as nx
+from itertools import combinations, chain
+from networkx.utils import pairwise, not_implemented_for
+
+__all__ = ['metric_closure', 'steiner_tree']
+
+
+@not_implemented_for('directed')
+def metric_closure(G, nodes, weight="weight"):
+    """  Returns the sub-graph of the metric closure of the graph G induced by
+        the nodes.
+
+    Parameters:
+    -----------
+    G : Networkx graph
+
+    nodes : List
+            The function returns the sub-graph of metric closure induced by the
+            nodes specified in this list.
+
+    Returns:
+    --------
+    Metric closure of G induced by nodes.
+
+    M : Networkx graph
+
+    Note: If parameter nodes contain all nodes of the graph G then the function
+        returns metric closure of the entire
+        graph.
+
+    """
+    M = nx.Graph()
+
+    for u, v in combinations(nodes, 2):
+        distance, path = nx.single_source_dijkstra(G, u, v, weight=weight)
+        M.add_edge(u, v, distance=distance[v], path=path[v])
+
+    return M
+
+
+@not_implemented_for('directed')
+def steiner_tree(G, terminal_nodes, weight='weight'):
+    """ Returns steiner tree of the graph induced by terminal_nodes.
+
+    Parameters:
+    -----------
+
+    G : Networkx graph
+
+    terminal_nodes : List
+                     A list of terminal nodes for which steiner tree is to be
+                     found.
+
+    Returns:
+    --------
+    The steiner tree for the given terminal nodes.
+    T : Networkx graph
+        Steiner tree.
+
+    Note: Steiner tree can be approximated by computing the minimum spanning
```

Comment:
This should go under a

```
Notes
-----
```
heading.

This section should also state the approximation guarantee provided by this algorithm. Ideally, it would also indicate the running time (and memory usage, if notable).

Also, lines in a paragraph should not be indented beyond four spaces. Indentation is used in the parameter list to define a [definition list][1] in reStructured Text format.

[1]: http://docutils.sourceforge.net/docs/user/rst/quickref.html#definition-lists